### DISCUSSION TITLE
Removing data incrementally

### DISCUSSION THREAD

**Discussion by jeffpeck10x** [visgl/deck.gl#6772] [Q&A] [answered]
The documentation describes an approach to asynchronously increment a Layer's `data`:
https://deck.gl/docs/developer-guide/performance#handle-incremental-data-loading

The comment explains:
```
// When a new chunk arrives, deck.gl only updates the sub buffers for the new rows
```

Would it be possible to similarly _remove_ data from the sub buffers that is no longer in view?

Consider a dataset with tens of millions of points and a limited zoom range such that only a million or so points max are within the viewport bounds at any given time. Incremental updates are appealing here, but only if there is also a way to remove the points that are no longer in view.

Is there a way to do this?

**Comment by ibgreen** [comment, marked as answer]
Well, the functionality you describe allows you to efficiently append rows to the end of data table. 

While it could certainly be extended in various ways (e.g. to drop rows from the beginning of the table when the table gets too long, in a FIFO type setup), you are looking for something different: a more sophisticated system that keep track of which chunks of rows need to be cut based on geospatial extents of the data continuously matched against changing viewports.

That type of chunking and extents tracking is exactly what the various TileLayers and tiled data formats were created for. They manage incremental loading, caching, unloading etc. So maybe just use those (perhaps with an adapter that serves up chunks tile data from your own in-memory/run-time processing, instead of loading pre-generated tile urls)?

**Reply by jeffpeck10x** [reply]
Interesting. Would something like a scatter-plot be possible with the TileLayer? I thought it was more for serving raster data?

**Reply by ibgreen** [reply]
The TileLayer renders GeoJSON by default (and uses a ScatterplotLayer for GeoJSON points) and can be customized.

**Reply by jeffpeck10x** [reply]
Amazing! Thank you for the suggestion. I would not have thought to use it with anything other than the BitmapLayer.
